zebra-chain/src/transaction/sighash.rs

```diff
@@ -0,0 +1,292 @@
+#![allow(dead_code, unused_variables)]
+use super::{Transaction, TransparentInput};
+use crate::{
+    parameters::ConsensusBranchId, serialization::ZcashSerialize, types::BlockHeight, Network,
+    NetworkUpgrade,
+};
+use blake2b_simd::Hash;
+use byteorder::{LittleEndian, WriteBytesExt};
+use std::io;
+
+const ZCASH_SIGHASH_PERSONALIZATION_PREFIX: &[u8; 12] = b"ZcashSigHash";
+const ZCASH_PREVOUTS_HASH_PERSONALIZATION: &[u8; 16] = b"ZcashPrevoutHash";
+const ZCASH_SEQUENCE_HASH_PERSONALIZATION: &[u8; 16] = b"ZcashSequencHash";
+const ZCASH_OUTPUTS_HASH_PERSONALIZATION: &[u8; 16] = b"ZcashOutputsHash";
+const ZCASH_JOINSPLITS_HASH_PERSONALIZATION: &[u8; 16] = b"ZcashJSplitsHash";
+const ZCASH_SHIELDED_SPENDS_HASH_PERSONALIZATION: &[u8; 16] = b"ZcashSSpendsHash";
+const ZCASH_SHIELDED_OUTPUTS_HASH_PERSONALIZATION: &[u8; 16] = b"ZcashSOutputHash";
+
+pub const SIGHASH_ALL: u32 = 1;
+const SIGHASH_NONE: u32 = 2;
+const SIGHASH_SINGLE: u32 = 3;
+const SIGHASH_MASK: u32 = 0x1f;
+const SIGHASH_ANYONECANPAY: u32 = 0x80;
```

Comment:
Should this be an enum used as a parameter for the sighash type?

Reply:
using an enum would require a ton of `as u32` conversions to be able to use the variants as masks.

Reply:
Ah, yeah, if the sighash mode is a bitflag we should use the bitflags crate.  We already use bitflags in the networking code so there's no cost to adding the dependency here, and it's much better. :+1: